lib/tar.ml

```diff
@@ -371,47 +368,52 @@ module Header = struct
 
   (** Unmarshal a header block, returning None if it's all zeroes *)
   let unmarshal ?level ?(extended = Extended.make ()) (c: Cstruct.t) : t option =
+    if Cstruct.length c <> length then invalid_arg "bad block size";
     let level = get_level level in
-    if allzeroes c then None
+    if Cstruct.equal c zero_block then None
```

Comment:
the issue I see is that `zero_block` is exposed from the interface, and a Cstruct.t, so any library can potentially mutate the data.

maybe it would be fine to not expose `zero_block`?